CMakeLists.txt

```diff
@@ -1,7 +1,7 @@
 # CMakeLists.txt
 
 # CMake setup
-cmake_minimum_required (VERSION 3.8)
+cmake_minimum_required (VERSION 3.11)
```

Comment:
I'm guessing this is one of the issues with the scripts.  At least the `ubuntu-latest` images are with CMake 3.10.

If this is the issue - then I'm happy with a CMake version bump as long as you update the install_dependencies.sh scripts to fix the CI loop.

Reply:
What do you think of https://github.com/marketplace/actions/get-cmake ?

Reply:
I don't object - its a public repo action with a suitable license.  As long as you configure it for the version of cmake shipped in Windows Visual Studio (3.16.x I think).  Ubuntu 20.04 and macOS via brew both ship 3.17.

But my further question is: Will it work with the docker container runs?

Reply:
Couldn't get that action to work so tried something that should have been easy. I'm stumped as to why Ubuntu bionic is failing - it is setting the path but then apparently ignoring it.

Still getting used to this CI, I thought that you would have to set it up as I am familiar with Hudson/Jenkins and latterly Travis.

Reply:
Am I right in thinking that all the other containers ship with 3.17 and only ubuntu:bionic ships with 3.10? So the manual install of 3.17 is only relevant for ubuntu:bionic? Is it possible that the export is not being picked up because the subsequent processes are separate from the process running install_dependencies.sh?

Reply:
@insidedctm Most of the containers were failing until I added the download of CMake (tar -xf and add to path) after which it was only bionic. So I've commented out traversing into third_party and unit_tests and rolled back the required CMake version. I know when I am beat.